nighthawk/source/client/factories_impl.cc

```diff
@@ -0,0 +1,66 @@
+#include "nighthawk/source/client/factories_impl.h"
+
+#include "common/stats/isolated_store_impl.h"
+
+#include "nighthawk/source/client/benchmark_client_impl.h"
+#include "nighthawk/source/common/platform_util_impl.h"
+#include "nighthawk/source/common/rate_limiter_impl.h"
+#include "nighthawk/source/common/sequencer_impl.h"
+#include "nighthawk/source/common/statistic_impl.h"
+#include "nighthawk/source/common/utility.h"
+
+namespace Nighthawk {
+namespace Client {
+
+OptionBasedFactoryImpl::OptionBasedFactoryImpl(const Options& options) : options_(options) {}
+
+BenchmarkClientFactoryImpl::BenchmarkClientFactoryImpl(const Options& options)
+    : OptionBasedFactoryImpl(options) {}
+
+BenchmarkClientPtr BenchmarkClientFactoryImpl::create(Envoy::Api::Api& api,
+                                                      Envoy::Event::Dispatcher& dispatcher,
+                                                      Envoy::Stats::Store& store,
+                                                      const Uri uri) const {
+  StatisticFactoryImpl statistic_factory(options_);
+
+  auto benchmark_client =
+      std::make_unique<BenchmarkClientHttpImpl>(api, dispatcher, store, statistic_factory.create(),
+                                                statistic_factory.create(), uri, options_.h2());
+  benchmark_client->setConnectionTimeout(options_.timeout());
+  benchmark_client->setConnectionLimit(options_.connections());
+  return benchmark_client;
+}
+
+SequencerFactoryImpl::SequencerFactoryImpl(const Options& options)
+    : OptionBasedFactoryImpl(options) {}
+
+SequencerPtr SequencerFactoryImpl::create(Envoy::TimeSource& time_source,
+                                          Envoy::Event::Dispatcher& dispatcher,
+                                          BenchmarkClient& benchmark_client) const {
+  StatisticFactoryImpl statistic_factory(options_);
+
+  RateLimiterPtr rate_limiter =
+      std::make_unique<LinearRateLimiter>(time_source, Frequency(options_.requests_per_second()));
+  SequencerTarget sequencer_target =
+      std::bind(&BenchmarkClient::tryStartOne, &benchmark_client, std::placeholders::_1);
```

Comment:
Nit: prefer lambdas, they are more readable and fewer corner cases.

Reply:
I'll ping you on slack for this one;